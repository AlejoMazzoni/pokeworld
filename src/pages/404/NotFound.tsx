import { Link } from "react-router-dom";
import styles from "./NotFount.module.scss";
import image404 from "../../assets/images/404-image.png";

const NotFound = () => {
  return (
    <div className={styles.notFoundContainer}>
      <div className={styles.textContainer}>
        <img src={image404} />
        <h3>Page not found.</h3>
        <p>
          It seems the page you're looking for doesn't exist.
          <br />
          <br />
          Click the button to return home
        </p>
      </div>
      <div className={styles.buttonContainer}>
        <Link to="/">
          <button>GO BACK</button>
        </Link>
      </div>
    </div>
  );
};

export default NotFound;
