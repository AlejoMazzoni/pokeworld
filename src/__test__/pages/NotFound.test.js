import renderer from "react-test-renderer";
import NotFound from "../../pages/404/NotFound";
import { HashRouter } from "react-router-dom";

describe("NotFound", () => {
  it("renders everything correctly", () => {
    const component = renderer.create(
      <HashRouter>
        <NotFound />
      </HashRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
