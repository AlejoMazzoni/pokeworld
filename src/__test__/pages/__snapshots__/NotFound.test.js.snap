// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders everything correctly 1`] = `
<div
  className="notFoundContainer"
>
  <div
    className="textContainer"
  >
    <div
      className="backdropFilter"
    />
    <img
      src="404-image.png"
    />
    <h3>
      Page not found.
    </h3>
    <p>
      It seems the page you're looking for doesn't exist.
      <br />
      <br />
      Click the button to return home
    </p>
  </div>
  <div
    className="buttonContainer"
  >
    <a
      href="#/"
      onClick={[Function]}
    >
      <div
        className="buttonContainer"
      >
        <button>
          GO BACK
        </button>
      </div>
    </a>
  </div>
</div>
`;
